syft/pkg/cataloger/dart/parse_pubspec_lock.go

```diff
@@ -89,6 +111,68 @@ func parsePubspecLock(_ context.Context, _ file.Resolver, _ *generic.Environment
 	return pkgs, nil, unknown.IfEmptyf(pkgs, "unable to determine packages")
 }
 
+// Look up the version range constraint for a given sdk name, if found,
+// and return its lowest supported version matching that constraint.
+//
+// The sdks and their constraints are defined in the pubspec.lock file, e.g.
+//
+//	sdks:
+//		dart: ">=2.12.0 <3.0.0"
+//		flutter: ">=3.24.5"
+//
+// and stored in the pubspecLock.Sdks map during parsing.
+//
+// Example based on the data above:
+//
+//	getSdkVersion("dart") -> "2.12.0"
+//	getSdkVersion("flutter") -> "3.24.5"
+//	getSdkVersion("undefined") -> error
+func (psl *pubspecLock) getSdkVersion(sdk string) (string, error) {
+	constraint, found := psl.Sdks[sdk]
+
+	if !found {
+		return "", fmt.Errorf("cannot find %s SDK", sdk)
+	}
+
+	return parseMinimumSdkVersion(constraint)
```

Comment:
the python cataloger does similar behavior for requirements.txt parsing (when a range is found). In that case we have a configuration item for the cataloger. Just mentioning from a consistency perspective, but ultimately I think that a config item probably is not needed in this case.

Reply:
I think the case is a bit different with Dart here.

As far as I can tell, `pubspec.lock` always lists exact versions for the packages (`pubspec.yaml` would be a different situation), only the SDK section in it uses ranges / minimum version. Making the version guessing a config option would potentially mean that anything related to the Flutter SDK, including Flutter itself, might be omitted from the SBOM. Not sure if that's a good idea :)